boards/mediatek/mt8188_adsp/board.yml

```diff
@@ -2,4 +2,4 @@ boards:
   - name: mt8188_adsp
     vendor: mediatek
     socs:
-      - name: mt8188_adsp
+      - name: mt8188
```

Comment:
Changes are OK but you could alternatively rename the board to mt8188, then the filenames would be mt8188_adps.dts as it was before, and means your minimal build string would be `west build -b mt8188//adps`. Please squash the changes to the socs/boards added in earlier commits into the earlier commits

Reply:
I'd really rather have it be a separate patch if it's OK with you, for symmetry with the older already-merged device that is being changed and just to have a record I know I can look back to.

And the name still creeps me out.  Again, this is **not** an integration for the whole laptop chip.  It's just audio firmware.

Reply:
How about if you flip the order of converting existing boards to the new format as the first commit then have the new commits adding the final board names in the commits without having a fixup commit at the end?

Reply:
Does that really rise to a -1?  It's a non-trivial rebase, and I personally like the order here as it makes the "complicated" change obvious.

Reply:
You are adding a commit https://github.com/zephyrproject-rtos/zephyr/pull/81154/commits/04ed839c1d989d752364f2259234aba4c85aa0ff and then fixing it in the next commit https://github.com/zephyrproject-rtos/zephyr/pull/81154/commits/eab814b5b8f4ff4b49635c7f41b3d8ac0c1a8f5f so yes. It's fine to have the commits in that order, but the first one should be changed to add the new boards with the correct names outright, then the second commit for updating existing boards, there is no point to adding a commit which has no use, is wrong and is fixed up in a later commit.

Reply:
I'm going to try one more time: I really don't want to do that, because if I do that then I'm not going to have a reference to go back to explaining how I did this the first time.  I understand you view that last commit as "obvious" renaming, but I'm not so talented and had to spend a lot of effort to fix things based on my pre-existing understanding of how zephyr board/soc integration works.  I'd like to know I can get back to it to see how it worked.

Basically, can you pretty-please remove your -1 here?  I really don't think commit order should be a merge-blocking kind of thing.  It's a big project and we all need to recognize that different people have different aesthetic senses of how a git log should look.

Reply:
We have a whole porting guide here: https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#hw-model-v2
We have an automatic porting script here: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/utils/board_v1_to_v2.py
We have a whole branch here showing conversions we did for in-tree boards: https://github.com/zephyrproject-rtos/zephyr/tree/collab-hwm
Why do we need a new board to be added that, as part of the same PR, adds the board wrongly in the first commit by not using the things specifically defined for hwmv2 and also should anyone want to bisect a failure for this board would result in them having to use a different board name if they get to the first commit?

If you think the zephyr documentation is lacking, you can open an issue about it or submit a PR improving it, if you want to reference to see how you ported something then you can host that yourself locally on or github or anywhere

Reply:
> adds the board wrongly in the first commit by not using the things specifically defined for hwmv2

Yeah, I don't want this to become a big monkey fight, but that's distressingly overstated.  In fact I checked: there's nothing in the existing docs that demands the use of a cpucluster, it just talks about what the syntax means.  The demand for particular structure was **you're personal design aesthetic**.  And... that's fine!  Your code, your model, you get to make demands like that.

But it's not correct to characterize the early patches as "incorrect" either, given that they matched the idioms used by many (probably most) existing boards in the tree.  So if you do make those demands, part of the job of being a good maintainer is staying with the PR as developers push back with questions and attempts at compromise and not walking away in pique.

Reply:
If there are other boards not complying then they should be fixed, these are not boards/socs I use so I don't know of any